src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -340,3 +340,60 @@ def load_secure_props(self) -> None:
                 f" with error '{error_msg}'",
                 SecurePropsNotFoundWarning,
             )
+
+    def __is_secure(self, json_path: str) -> bool:
+        """
+        Check whether the given JSON path corresponds to a secure property.
+
+        Parameters:
+            json_path (str): The JSON path of the property to check.
+
+        Returns:
+            bool: True if the property should be stored securely, False otherwise.
+        """
+        profile_name, property_name = json_path.split(".")[1:]
+        profile = self.find_profile(profile_name, self.profiles)
+        if profile and profile.get("secure"):
+            return property_name in profile["secure"]
+        return False
+
+    def set_property(self, json_path, value, secure=False):
```

Comment:
In order to match the behavior of the Node.js SDK, I think the default value of `secure` should be None.

* If `secure` is True, then we want to add the property name to the `secure` array and store the property value in the vault.
* If `secure` is False, then we want to remove the property name from the `secure` array and store the property value in plain text in the JSON file.
* If `secure` is None, then we want to detect whether the property is in the `secure` array. If so we treat it as secure, and if not we treat it as plain text.